frontend/app/assets/javascripts/spree/frontend/currency_selector.js

```diff
@@ -0,0 +1,16 @@
+$(function() {
```

Comment:
I was thinking we could remove all the JS, including the ajax call and the new route/controller action if we use [`data-url`](https://guides.rubyonrails.org/working_with_javascript_in_rails.html#data-url-and-data-params) in combination with the select tag instead.

For the optional confirm message, maybe we can just add `data-confirm` on the element conditionally checking if the order has items directly in the view.

```erb
          <%= select_tag(
            :switch_to_currency,
            options_for_select(
              supported_currencies,
              selected: current_currency
            ),
            class: 'form-control'),
            data: {
              url: spree.select_currency_path,
              method: :put,
              confirm: current_order_in_progress? I18n.t('spree.currency_switch_confirm') : false
            } %>
```

I didn't test it and not 100% sure it works as is. Thoughts?

Reply:
It would change the behavior in two ways:

1. It would change the switch currency action to work asynchronously.
2. We would lose the asynchronous call to check whether the cart is empty.

About 1, I don't like the idea because the frontend architecture follows good old sync requests-responses, so we should add complexity to update the DOM and display the new prices. It would also be inconsistent with how locale switching works.

About 2, using the `data-confirm` value, which is set when the HTML is loaded, would make it possible that we missed products added on another browser tab.

Reply:
You are right, I didn't consider we want this request to be sync. In terms of `data-confirm`, personally, I think that for the sake of simplicity, removing the real-time check is a good trade-off. I think we can ship a basic version when someone has multiple tabs open and doesn't see the popup if they change the currency in some of those tabs. If that's important for some businesses, it's very easy to add that portion of code in their app.

Reply:
Probably it makes sense... I removed the async check in a new commit. Please, take a look at it. As we still need to keep the current currency as the selected option when the action is aborted, we can't rely on Rails `data-confirm` magic attribute, so it's still done through the JS file.